### DISCUSSION TITLE
Wake word detection

### DISCUSSION THREAD

**Discussion by fanti08** [deepgram/community#582] [General help] [answered]
Yesterday I managed to implement a wake-word (and also a stop-word) detection system in my code - I'm building an AI voice assistant, and after testing with various model (SpeechRecognition by Google, Assembly AI, whisper etc.), I figured that Deepgram is the best in terms of accurancy, speed and cost. But I can't find a way to cost-effectivly implement like a 24/7 hotword detection system.
If anybody could give me any help that would be really appreciated, thanks.
P.S.: I also found another discussion from May where the same thing was asked, but it ended up being closed because the guy opted for a non-vocal way to start his assistant.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by nikolawhallon** [comment, marked as answer]
Deepgram doesn't offer wake-word detection specifically (that is, for example, a lightweight model which is always running tuned just on the wake-word), this can be accomplished by doing speech-to-text 24/7, but maybe that is expensive for your use-case.

If you want to keep a connection to Deepgram open 24/7, but only stream audio for part of that duration, we have a `KeepAlive` feature which can help do this:

https://developers.deepgram.com/reference/listen-live#stream-keepalive

If you pair that with some technique for the wake-word detection (which isn't simply transcribing in the background 24/7), that could work.

**Reply by fanti08** [reply]
> If you pair that with some technique for the wake-word detection (which isn't simply transcribing in the background 24/7), that could work.

And what would that look like if I may ask ?

**Reply by nikolawhallon** [reply]
I imagine one of the more efficient ways to handle this would be to locally custom train a super tiny model trained on a single word - that would give the wake-word detection needed, while being a small and fast (and cheap) model to run.

It's a reasonable idea in my opinion for Deepgram to have a model like this, but since the actual word desired might depend on use-case/client/user/etc, it might be tough to provide this kind of light-weight model for any arbitrary word (but I can imagine support for a "hey Deepgram" wake-word one day, for example).
